### PR TITLE
abc-346 user password fix

### PR DESCRIPTION
**Pull request by Mil-m**

### CODE REVIEW

**Comment by Mil-m** [comment]
possibly we also have to generate error in case of the incorrect password
it's weird but I don't see it on the BE side by default

**Comment by Mil-m** [comment]
in addition, in case of creating TNA Project, the additional participant user will be created without any password (that is not correct for our Django admin requirements)
although this is not relevant to this task and PR

Screenshot 2024-06-06 at 14 08 37

Screenshot 2024-06-06 at 14 12 15

**Comment by Alexey-ebi** [comment]
For all additional features/bugs report please create JIRA tickets within current epic

File: eurofaang_ri_be/eurofaang_ri_be/settings.py [hunk 1]
```diff
@@ -113,6 +113,13 @@
     },
 ]
 
+PASSWORD_HASHERS = [
+    "django.contrib.auth.hashers.PBKDF2PasswordHasher",
+    "django.contrib.auth.hashers.PBKDF2SHA1PasswordHasher",
+    "django.contrib.auth.hashers.Argon2PasswordHasher",
+    "django.contrib.auth.hashers.BCryptSHA256PasswordHasher",
+    "django.contrib.auth.hashers.ScryptPasswordHasher",
+]
```
Comment (Alexey-ebi) [inline eurofaang_ri_be/eurofaang_ri_be/settings.py]: I've just tested it and looks like it doesn't have any effect on the password hashing logic. Do we really need to have it here?

File: eurofaang_ri_be/users/models.py [hunk 2]
```diff
@@ -36,6 +36,15 @@ def __str__(self):
     organization_country = models.CharField(choices=COUNTRIES_CHOICES, blank=True, null=True)
     role = models.CharField(max_length=2, choices=ROLE_CHOICES, default='PI')
 
+    def save(self, *args, **kwargs):
+        if self.pk is None and self.password:
+            self.set_password(self.password)
+        elif self.pk is not None:
```
Comment (Alexey-ebi) [inline eurofaang_ri_be/users/models.py]: can this be simplified to just "if else", what happens if self.password is None?

**Review by Alexey-ebi — CHANGES_REQUESTED** [review]

File: eurofaang_ri_be/eurofaang_ri_be/settings.py (hunk 1 above)
Comment (Mil-m) [reply eurofaang_ri_be/eurofaang_ri_be/settings.py]: we don't really need it as this algorithm is used by default

https://docs.djangoproject.com/en/5.0/topics/auth/passwords/

I thought it was make sense to specify it more explicitly in the settings

**Review by Alexey-ebi — APPROVED** [review]
LGTM

**Pull request merged**